src/rgw/rgw_formats.cc

```diff
@@ -167,10 +167,12 @@ void RGWFormatter_Plain::dump_format_va(std::string_view name, const char *ns, b
     eol = "";
   wrote_something = true;
 
-  if (use_kv && !entry.is_array)
-    write_data("%s%.*s: %s", eol, name.size(), name.data(), buf);
-  else
+  if (use_kv && !entry.is_array) {
+    write_data("%s%.*s: %s", eol, (int)(name.size()), name.data(), buf);
```

Comment:
please dont use c-style casts.
some reasons:
* they can explicitly override constness
* they are hard to notice and grep